src/msix/pack/ContentTypeWriter.cpp

```diff
@@ -0,0 +1,90 @@
+//
+//  Copyright (C) 2019 Microsoft.  All rights reserved.
+//  See LICENSE file in the project root for full license information.
+// 
+
+#include "XmlWriter.hpp"
+#include "ContentTypeWriter.hpp"
+
+#include <map>
+#include <algorithm>
+
+namespace MSIX {
+
+    /*
+    <Types xmlns="http://schemas.openxmlformats.org/package/2006/content-types">
+      <Default ContentType="image/png" Extension="png"/>
+      <Default ContentType="application/x-msdownload" Extension="dll"/>
+      <Default ContentType="application/vnd.ms-appx.manifest+xml" Extension="xml"/>
+      <Override ContentType="application/vnd.ms-appx.blockmap+xml" PartName="/AppxBlockMap.xml"/>
+      <Override ContentType="application/vnd.ms-appx.signature" PartName="/AppxSignature.p7x"/>
+      <Override ContentType="application/vnd.ms-pkiseccat" PartName="/AppxMetadata/CodeIntegrity.cat"/>
+    </Types>
+    */
+
+    static const char* typesElement = "Types";
+    static const char* typesNamespace = "http://schemas.openxmlformats.org/package/2006/content-types";
+    static const char* defaultElement = "Default";
+    static const char* contentTypeAttribute = "ContentType";
+    static const char* extensionAttribute = "Extension";
+    static const char* overrideElement = "Override";
+    static const char* partNameAttribute = "PartName";
+
+    // <Types xmlns="http://schemas.openxmlformats.org/package/2006/content-types">
+    ContentTypeWriter::ContentTypeWriter() : m_xmlWriter(XmlWriter(typesElement))

```

Comment:
>XmlWriter [](start = 57, length = 9)

Not needed.

Reply:
I don't see why this isn't needed. XmlWriter doesn't have a default constructor which is intentional

---
In reply to: [278648822](https://github.com/Microsoft/msix-packaging/pull/95#discussion_r278648822) [](ancestors = 278648822)